src/utils/staking.js

```diff
@@ -0,0 +1,278 @@
+    }
+
+    async withdraw(useLockup) {
+        const { lockupId } = await this.getLockup()
+        const withdraw_all_from_staking_pool = await this.signAndSendTransaction(lockupId, [
+            functionCall('withdraw_all_from_staking_pool', {}, GAS_STAKE)
+        ])
+        console.log('withdraw_all_from_staking_pool', withdraw_all_from_staking_pool)
+        if (withdraw_all_from_staking_pool[0] === false) {
+            throw new Error('Unable to withdraw pending balance from validator')
+        }
+    }
+
+    // helpers for lockup
+
+    async lockupStake(lockupId, validatorId, amount) {
+        const deposit_and_stake = await this.signAndSendTransaction(lockupId, [
+            functionCall('deposit_and_stake', { amount }, GAS_STAKE)
+        ])
+        console.log('deposit_and_stake', deposit_and_stake)
+        return deposit_and_stake
+    }
+
+    async lockupSelect(validatorId, lockupId, unselect = false) {
+        if (unselect) {
+            try {
+                const unselect_staking_pool = await this.signAndSendTransaction(lockupId, [
+                    functionCall('unselect_staking_pool', {}, GAS_STAKE)
+                ])
+                console.log('unselect_staking_pool', unselect_staking_pool)
+            } catch (e) {
+                console.warn('Problem unselecting validator', validatorId, e)
+            }
+        }
+        try {
+            const select_staking_pool = await this.signAndSendTransaction(lockupId, [
+                functionCall('select_staking_pool', { staking_pool_account_id: validatorId }, GAS_STAKE)
+            ])
+            console.log('select_staking_pool', select_staking_pool)
+        } catch (e) {
+            console.warn('Problem selecting validator', validatorId, e)
+            throw new Error('Cannot select validator')
+        }
+    }
+
+    async getLockup() {
+        const accountId = this.wallet.accountId
+
+        let lockupId
+        if (process.env.REACT_APP_NETWORK_ID === 'mainnet') {
+            lockupId = await getLockupId(accountId)
+        } else {
+            // TODO remove for main release - for testnet and betanet
+            lockupId = `testinglockup.${accountId}`
+            try {
+                await (await new nearApiJs.Account(this.wallet.connection, lockupId)).state()
+            } catch(e) {
+                lockupId = testLockup[accountId] ? testLockup[accountId] : await getLockupId(accountId)
+            }
+        }
+
+        const contract = await this.getContractInstance(lockupId, lockupMethods)
+        return { contract, lockupId, accountId }
+    }
+
+    async getContractInstance(contractId, methods) {
+        try {
+            await (await new nearApiJs.Account(this.wallet.connection, contractId)).state()
+            return await new nearApiJs.Contract(this.wallet.getAccount(), contractId, { ...methods })
+        } catch(e) {
+            console.warn(e)
+            throw Error('No lockup contract for account')
```

Comment:
I don't think this is true unless you check for specific error about account not being found